README.md

```diff
@@ -30,22 +30,41 @@ const options = [{
-- **`displayCaret`** controls whether the dropdown caret should be present (default: `false`)
-- **`closeOnSelect`** controls whether selecting an option in the dropdown menu should close it (default: `false`)
-- **`openViaButton`** controls whether clicking on the main button (not including the caret) should open the dropdown menu (default: `true`)
-- **`openViaCaret`** controls whether clicking on the main button's caret (if present) should open the dropdown menu (default: `true`)
-- **`includeSelectedOption`** controls whether the currently-selected option is included in the dropdown menu (default: `false`)
+- **`open`** controls whether the dropdown menu is currently visible
+  - default: `false`
+- **`displayCaret`** controls whether the dropdown caret should be present
+  - default: `false`
+- **`closeOnSelect`** controls whether selecting an option in the dropdown menu should close it
+  - default: `false`
+- **`openViaButton`** controls whether clicking on the main button (not including the caret) should open the dropdown menu
+  - default: `true`
+- **`openViaCaret`** controls whether clicking on the main button's caret (if present) should open the dropdown menu
+  - default: `true`
+- **`includeSelectedOption`** controls whether the currently-selected option is included in the dropdown menu
+  - default: `false`
 - **`options`** specifies the list of items available in the dropdown menu, using the structure shown above. Only `id` and `label` are required.
-- **`selectedOption`** is a string specifying the `id` of the currently-selected option (default: `''`)
-- **`optionGroup`** is a string describing the dropdown's purpose, accessible to screen-readers only (default: `'options'`)
+  - default: `[]`
+  - Other menu option types can be used by setting `isCustomList` to `true` and using the `menu-slot` slot with a custom list component.
+- **`selectedOption`** is a string specifying the `id` of the currently-selected option
+  - default: `''`
+- **`optionGroup`** is a string describing the dropdown's purpose, accessible to screen-readers only
+  - default: `options`
+-  **`isCustomList`** replaces default option list with slot `menu-slot` allowing a custom list component to be used with different option type and behavior
+   - default: `false`
 
 #### `<ia-dropdown>` slots
 - `slot="dropdown-label"` the main button that opens the dropdown menu.
 - `slot="caret-up"` replaces default up caret svg.
   - Use both `slot="caret-up"` and `class="caret-up-svg"` in the `<svg>`.
 - `slot="caret-down"` replaces default down caret svg.
   - Use both `slot="caret-down"` and `class="caret-down-svg"` in the `<svg>`.
+- `slot="menu-slot"` allows replacing default dropdown menu option type and behavior.
+
+#### `<ia-dropdown>` events
+Listens for:
+- `closeDropdown` closes the dropdown menu
+Dispatches:
+- `optionSelected` when an option is selected in the dropdown menu
```

Comment:
Thanks for cleaning up the README & documenting new features!

Reply:
You're welcome!